perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python module for setting up the dataset and tables in BigQuery.
+
+This python module creates the dataset and the tables that will store experiment
+configurations and metrics data in BigQuery.
+
+Note:
+  Make sure BigQuery API is enabled for the project
+"""
+import sys
+from google.cloud import bigquery
+
+
+class ExperimentsGCSFuseBQ:
+  """
+    Class to create and interact with Bigquery dataset and tables for storing
+    experiments configurations and their results.
+
+    Attributes:
+      project_id (str): The GCP project in which dataset and tables will be created
+      dataset_id (str): The name of the dataset in the project that will store the tables
+      bq_client (Optional[google.cloud.bigquery.client.Client]): The client for interacting with Bigquery.
+                                                                 Default value is bigquery.Client(project=project_id).
+  """
+
+  CONFIGURATION_TABLE_ID = 'experiment_configuration'
+  FIO_TABLE_ID = 'read_write_fio_metrics'
+  VM_TABLE_ID = 'read_write_vm_metrics'
+  LS_TABLE_ID = 'list_metrics'
+
+  def __init__(self, project_id, dataset_id, bq_client=None):
+    if bq_client is None:
+      self.client = bigquery.Client(project=project_id)
+    self.project_id = project_id
+    self.dataset_id = dataset_id
+
+  @property
+  def dataset_ref(self):
+    """
+      Gets the reference of the dataset
+
+      Returns:
+        google.cloud.bigquery.dataset.Dataset: The retrieved dataset object
+    """
+    return self.client.get_dataset(self.dataset_id)
+
+  def _execute_query_and_check_for_error(self, query) -> QueryJob:
+    """Executes the query in BigQuery and raises an exception if query
+       execution could not be completed.
+
+    Args:
+      query (str): Query that will be executed in BigQuery.
+
+    Raises:
+      Aborts the program if error is encountered while executing the query.
```

Comment:
nit: Follow the style guide https://google.github.io/styleguide/pyguide#383-functions-and-methods. 
So should be something like
Raises:
   Exception: Error while executing the query.
 
 E.g. https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/perfmetrics/scripts/load_tests/python/load_test.py#L187